adr: replace starboard-operator with trivy-operator (#556)
elastisys/compliantkubernetes: docs/adr/0038-replace-starboard-operator-with-trivy-operator.md

# Replace the starboard-operator with the trivy-operator

* Status: accepted
* Deciders: arch meeting
* Date: 2023-03-30

## Context and Problem Statement

> The maintainers of starboard https://github.com/aquasecurity/starboard/discussions/1173 in favor of [trivy kubernetes](https://aquasecurity.github.io/trivy/latest/docs/kubernetes/cli/scanning/) with [trivy operator](https://github.com/aquasecurity/trivy-operator). They will no longer make any bigger changes to starboard operator. They announced the change in march.

Reviewer: ```suggestion
The maintainers of starboard [deprecated it](https://github.com/aquasecurity/starboard/discussions/1173) in favor of [trivy kubernetes](https://aquasecurity.github.io/trivy/latest/docs/kubernetes/cli/scanning/) with [trivy operator](https://github.com/aquasecurity/trivy-operator). They will no longer make any bigger changes to starboard operator. They announced the change in march.
```

Author: Fixed
Thank you!

> We currently use starboard operator for scanning images with trivy and for running the CIS kubernetes benchmark with kubebench. Trivy operator has support for scanning images but does not currently support running the CIS kubernetes benchmark. They plan on adding support for something like kube-bench, but written specifically for trivy, see https://github.com/aquasecurity/trivy-operator/issues/141 and https://github.com/aquasecurity/trivy/issues/2198.

Reviewer: The part about trivy not including CIS benchmark is no longer true.
```suggestion
We currently use starboard operator for scanning images with trivy and for running the CIS kubernetes benchmark with kubebench. Trivy operator has support for scanning images and running a version of the CIS kubernetes benchmark. 
```

Author: Changed
Thx
